sql: correct error code for CopyFail message

otan opened the pull request:
Resolves #81559.

Release note (bug fix): Previously, cancelling COPY commands would have
an XXUUU error, instead of 57014. This is now rectified.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/81564)
<!-- Reviewable:end -->

otan committed:
sql: correct error code for CopyFail message

Release note (bug fix): Previously, cancelling COPY commands would have
an XXUUU error, instead of 57014. This is now rectified.

rafiss reviewed (approved):
nice!

otan commented:
bors r=rafiss
